src/vs/workbench/contrib/notebook/browser/contrib/find/notebookFindReplaceWidget.ts

```diff
@@ -345,6 +345,20 @@ export abstract class SimpleFindReplaceWidget extends Widget {
 
 		this._domNode = document.createElement('div');
 		this._domNode.classList.add('simple-fr-find-part-wrapper');
+
+		if (this._notebookEditor.notebookOptions.getLayoutConfiguration().globalToolbar) {
+			this._domNode.style.top = '26px';
+		}
+		this._register(this._configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(NotebookSetting.globalToolbar)) {
+				if (this._notebookEditor.notebookOptions.getLayoutConfiguration().globalToolbar) {
+					this._domNode.style.top = '26px';
+				} else {
+					this._domNode.style.top = '0';
+				}
```

Comment:
Since this is doing the same thing, this is a perfect use case for `runAndSubscribe` which will let you remove the upper if all together. Then `e` will be `| undefined` so change the condition to `if (!e || e.affects...)`

Reply:
How would this remove the upper condition? Would it not just add the `!e ||` to the beginning of it?

I still need to check the value of the setting being on/off, or is the assumption that it toggles the setting so just invert the current value and that would remove one of the conditions?